lighthouse-core/test/gather/gather-runner-test.js

```diff
@@ -7,39 +7,78 @@
-const {createMockSendCommandFn, createMockOnceFn} = require('./mock-commands.js');
-const {
+import {jest} from '@jest/globals';
+import Gatherer from '../../gather/gatherers/gatherer.js';
+// import GathererRunner_ from '../../gather/gather-runner.js';
+import {strict as assert} from 'assert';
+// import Config from '../../config/config.js';
+import unresolvedPerfLog from './../fixtures/unresolved-perflog.json';
+import LHError from '../../lib/lh-error.js';
+import networkRecordsToDevtoolsLog from '../network-records-to-devtools-log.js';
+// import Driver from '../../gather/driver.js';
+import Connection from '../../gather/connections/connection.js';
+import {createMockSendCommandFn, createMockOnceFn} from './mock-commands.js';
+import {
+  makeMocksForGatherRunner,
   makeParamsOptional,
   makePromiseInspectable,
   flushAllTimersAndMicrotasks,
-} = require('../test-utils.js');
-
-const GatherRunner = {
-  afterPass: makeParamsOptional(GatherRunner_.afterPass),
-  beginRecording: makeParamsOptional(GatherRunner_.beginRecording),
-  collectArtifacts: makeParamsOptional(GatherRunner_.collectArtifacts),
-  endRecording: makeParamsOptional(GatherRunner_.endRecording),
-  initializeBaseArtifacts: makeParamsOptional(GatherRunner_.initializeBaseArtifacts),
-  loadPage: makeParamsOptional(GatherRunner_.loadPage),
-  run: makeParamsOptional(GatherRunner_.run),
-  runPass: makeParamsOptional(GatherRunner_.runPass),
-  setupDriver: makeParamsOptional(GatherRunner_.setupDriver),
-  // Spies that should have mock implemenations most of the time.
-  assertNoSameOriginServiceWorkerClients: jest.spyOn(GatherRunner_,
-    'assertNoSameOriginServiceWorkerClients'),
+  fnAny,
+} from '../test-utils.js';
+import fakeDriver from './fake-driver.js';
+import {createCommonjsRefs} from '../../scripts/esm-utils.js';
+
+const {require} = createCommonjsRefs(import.meta);
+
+/**
+ * Same as jest.requireMock(), but uses `any` instead of `unknown`.
+ * @param {string} moduleName
+ * @return {any}
+ */
+const requireMockAny = (moduleName) => {
+  return jest.requireMock(moduleName);
 };
 
+makeMocksForGatherRunner();
+
+function createTypeHackedGatherRunner() {
+  return {
+    afterPass: makeParamsOptional(GatherRunner_.afterPass),
+    beginRecording: makeParamsOptional(GatherRunner_.beginRecording),
+    collectArtifacts: makeParamsOptional(GatherRunner_.collectArtifacts),
+    endRecording: makeParamsOptional(GatherRunner_.endRecording),
+    initializeBaseArtifacts: makeParamsOptional(GatherRunner_.initializeBaseArtifacts),
+    loadPage: makeParamsOptional(GatherRunner_.loadPage),
+    run: makeParamsOptional(GatherRunner_.run),
+    runPass: makeParamsOptional(GatherRunner_.runPass),
+    setupDriver: makeParamsOptional(GatherRunner_.setupDriver),
+    // Spies that should have mock implemenations most of the time.
+    assertNoSameOriginServiceWorkerClients: jest.spyOn(GatherRunner_,
+      'assertNoSameOriginServiceWorkerClients'),
+  };
+}
+
```

Comment:
This is the most complex example of the dynamic requirement workaround required for mocks to work.